src/osp/framework/framework.h

```diff
@@ -336,22 +337,26 @@ struct Framework
             // Access struct members of out.pl and out.di as if they're arrays, in order to write
             // DataIds and PipelineIds to them.
 
+            //Using this reference to avoid strict aliasing UB : https://gist.github.com/shafik/848ae25ee209f698763cffee272a58f8
+            //Apparently the memcpy calls get optimized away. 
+
             FeatureInterface const &rInterface = m_fiinstData[fiId];
+
             if constexpr ( ! std::is_empty_v<typename FI_T::Pipelines> )
             {
                 auto const plMembers = arrayCast<PipelineDefBlank_t>(  arrayCast<std::byte>( arrayView(&out.pl, 1) )  );
                 for (std::size_t i = 0; i < plMembers.size(); ++i)
                 {
-                    plMembers[i].m_value = rInterface.pipelines[i];
+                    memcpy(&plMembers[i].m_value,  &rInterface.pipelines[i], sizeof(PipelineId));
```

Comment:
minor: std::memcpy. iirc, this is C's memcpy from `<string.h>`, instead of `<cstring>`